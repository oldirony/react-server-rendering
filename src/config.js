export default {
	routes: {
		index: '/',
		addNewIdea: '/ideas/new',
		viewIdea: (id =':id') =>`/ideas/${id}`,
		editIdea: (id =':id') =>`/ideas/${id}/edit`,
		testApi: '/test-api',
		newIdeaApi: '/api/new-idea/',
		getIdeasApi: '/api/get-ideas/'
	}
}