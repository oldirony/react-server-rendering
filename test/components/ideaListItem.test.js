import React from 'react';
import IdeaListItem from '../../src/components/idea-list-item/idea-list-item';
import {shallow} from 'enzyme';
import {wrapper} from '../helper';
import classes from '../../src/components/idea-list-item/idea-list-item.scss';
import config from '../../src/config';

const props = {
	title: 'Idea title',
	timestamp: "2017-05-31T11:56:27.286Z",
	isCompleted: false,
	content: 'Lorem ipsum',
	sections: []
}


global.confirm = () => true


test('It renders correctly', ()=>{
	const ideaListItem = shallow(<IdeaListItem {...props} />);
	expect(ideaListItem).toMatchSnapshot();
});

describe('Behaviour', ()=>{
	const renderItem = (customProps={})=>mount(wrapper(<IdeaListItem {...props} {...customProps}/>));


	test('Get completed class', ()=>{
		const ideaListItem = renderItem({isCompleted:true});
		expect(ideaListItem.find(`.${classes.item}`).hasClass(classes.isCompleted)).toBeTruthy();
	});

	test('Renders section number', ()=>{
		const section = {
			title: 'Section title',
			content: 'My content here',
			isCompleted: false
		}
		let ideaListItem = renderItem({sections:[section, section]});
		expect(ideaListItem.find(`.${classes.completedSections}`).text()).toEqual('0 / 2');

		ideaListItem = renderItem({sections:[section, {...section, isCompleted:true}]});
		expect(ideaListItem.find(`.${classes.completedSections}`).text()).toEqual('1 / 2');

	});

	test('Has Link for view', ()=>{
		const ideaListItem = renderItem({isCompleted:true});
		expect(ideaListItem.find(`a`).node.href).toEqual(config.routes.viewIdea());
	});

	test('Test update button', ()=>{
		const handleUpdateClick = jest.fn();
		const ideaListItem = renderItem({handleUpdateClick});
		ideaListItem.find(`.${classes.buttons}`).find('button').first().simulate('click');

		expect(handleUpdateClick).toBeCalled();
	});

	test('Test delete button', ()=>{
		const handleDeleteClick = jest.fn();
		const ideaListItem = renderItem({handleDeleteClick});
		ideaListItem.find(`.${classes.buttons}`).find('button').last().simulate('click');

		expect(handleDeleteClick).toBeCalled();
	});


})