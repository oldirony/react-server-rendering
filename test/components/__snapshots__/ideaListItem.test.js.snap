// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
<li
  className="item"
>
  <Link
    className="link"
    replace={false}
    to="/ideas/:id"
  >
    <h2
      className="title"
    >
      Idea title
    </h2>
    <time
      className="date"
    >
      12:56 
      31-4-2017
    </time>
    <div
      className="completedSections"
    >
      0
       / 
      0
    </div>
    <div
      className="buttons"
    >
      <button
        className="vanilla withIcon"
        onClick={[Function]}
      >
        <svg
          className="icon"
        >
          <use
            xlinkHref="#checkbox-unchecked"
          />
        </svg>
        completed
      </button>
      <button
        className="vanilla withIcon"
        onClick={[Function]}
      >
        <svg
          className="icon"
        >
          <use
            xlinkHref="#bin"
          />
        </svg>
      </button>
    </div>
  </Link>
</li>
`;

exports[`Structure Get completed class 1`] = `
<li
  className="item isCompleted"
>
  <Link
    className="link"
    replace={false}
    to="/ideas/:id"
  >
    <h2
      className="title"
    >
      Idea title
    </h2>
    <time
      className="date"
    >
      12:56 
      31-4-2017
    </time>
    <div
      className="completedSections"
    >
      0
       / 
      0
    </div>
    <div
      className="buttons"
    >
      <button
        className="vanilla withIcon"
        onClick={[Function]}
      >
        <svg
          className="icon"
        >
          <use
            xlinkHref="#checkbox-checked"
          />
        </svg>
        completed
      </button>
      <button
        className="vanilla withIcon"
        onClick={[Function]}
      >
        <svg
          className="icon"
        >
          <use
            xlinkHref="#bin"
          />
        </svg>
      </button>
    </div>
  </Link>
</li>
`;

exports[`Structure Renders section number 1`] = `
<li
  className="item"
>
  <Link
    className="link"
    replace={false}
    to="/ideas/:id"
  >
    <h2
      className="title"
    >
      Idea title
    </h2>
    <time
      className="date"
    >
      12:56 
      31-4-2017
    </time>
    <div
      className="completedSections"
    >
      0
       / 
      2
    </div>
    <div
      className="buttons"
    >
      <button
        className="vanilla withIcon"
        onClick={[Function]}
      >
        <svg
          className="icon"
        >
          <use
            xlinkHref="#checkbox-unchecked"
          />
        </svg>
        completed
      </button>
      <button
        className="vanilla withIcon"
        onClick={[Function]}
      >
        <svg
          className="icon"
        >
          <use
            xlinkHref="#bin"
          />
        </svg>
      </button>
    </div>
  </Link>
</li>
`;

exports[`Structure Renders section number 2`] = `
<li
  className="item"
>
  <Link
    className="link"
    replace={false}
    to="/ideas/:id"
  >
    <h2
      className="title"
    >
      Idea title
    </h2>
    <time
      className="date"
    >
      12:56 
      31-4-2017
    </time>
    <div
      className="completedSections"
    >
      1
       / 
      2
    </div>
    <div
      className="buttons"
    >
      <button
        className="vanilla withIcon"
        onClick={[Function]}
      >
        <svg
          className="icon"
        >
          <use
            xlinkHref="#checkbox-unchecked"
          />
        </svg>
        completed
      </button>
      <button
        className="vanilla withIcon"
        onClick={[Function]}
      >
        <svg
          className="icon"
        >
          <use
            xlinkHref="#bin"
          />
        </svg>
      </button>
    </div>
  </Link>
</li>
`;
